### ISSUE TITLE
Build is broken on master

### ISSUE THREAD

**Issue by celskeggs** [sipb/homeworld#345]
See [CI 246](https://circleci.com/gh/sipb/homeworld/246):

```
$ #!/bin/bash -eo pipefail
echo "make -C upstream-check/ -j2 verify && killall gpg-agent" | HOMEWORLD_CHROOT="$HOME/autobuild-chroot" USER="circleci" ./enter-chroot-ci.sh

make: Entering directory '/homeworld/building/upstream-check'
[gpg] preparing keyring keys/acbuild.ring
[gpg] preparing keyring keys/coreos-app.ring
[https] fetching https://alpha.release.core-os.net/amd64-usr/1478.0.0/coreos_production_pxe_image.cpio.gz
[https] fetching https://alpha.release.core-os.net/amd64-usr/1478.0.0/coreos_production_pxe_image.cpio.gz.sig
[gpg] preparing keyring keys/coreos-image.ring
gpg: cannot open '/dev/tty': No such device or address
Makefile:48: recipe for target 'keys/coreos-image.ring' failed
make: *** [keys/coreos-image.ring] Error 2
make: *** Waiting for unfinished jobs....
rm https/alpha.release.core-os.net/amd64-usr/1478.0.0/coreos_production_pxe_image.cpio.gz.sig https/alpha.release.core-os.net/amd64-usr/1478.0.0/coreos_production_pxe_image.cpio.gz
make: Leaving directory '/homeworld/building/upstream-check'
Exited with code 2
```

It's odd that this is broken, because the relevant PR passed CircleCI tests. Maybe something changed about the environments that CircleCI is using?

**Comment by ichung** [comment]
@krawthekrow you investigated this, right?  Could you post your explanation about how this relates to the debian kernel bump?

**Comment by krawthekrow** [comment]
For broader context, there are two problems related to CI failing.

The first is something to do with GPG requesting interactive input. I'm not sure exactly what's causing it to do that since it's really hard to do any testing without a tty, but adding the "--batch" flag (as is recommended when calling GPG non-interactively) fixes this.

The second is the Debian version bump, which causes the upstream check to fail since it has to download from the official repository to check the checksum. Since Debian only keeps the latest version of their binaries in the repository, when they bump the version, trying to download the older version causes a 404.

Bumping the Debian version on our side revealed a bug in Spire, which was that the additional files we add to the initramfs weren't being compressed before being appended to initrd.gz. It looks like the latest version of Linux isn't able to read these files.

Currently the fixes have been implemented, I just need to figure out how to autodeploy with the new user grant system to complete the PR.

**Comment by ichung** [comment]
Fixed in #348
